nifi-nar-bundles/nifi-splunk-bundle/nifi-splunk-processors/src/main/java/org/apache/nifi/processors/splunk/QuerySplunkIndexingStatus.java

```diff
@@ -234,4 +241,14 @@ private String generateContent(final Map<Long, FlowFile> undetermined) throws IO
             return Optional.empty();
         }
     }
+
+    private static Optional<Boolean> extractBoolean(final String value) {
+        return Objects.nonNull(value) ? Optional.of(value).map(Boolean::valueOf) : Optional.empty();
+    }
+
+    private FlowFile setAckCheckedToTrue(final ProcessSession session, final FlowFile flowFile) {
+        final Map<String, String> attributes = new HashMap<>(flowFile.getAttributes());
```

Comment:
If you care about the HashMap size (which is a good thing :) ) add 1 to the flowFile.getAttributes() already, as you will extend the map. ;)